google/gax/__init__.py

```diff
@@ -464,10 +468,14 @@ def __iter__(self):
         return self
 
     def next(self):
+        """For Python 2.7 compatibility; see __next__."""
+        return self.__next__()
+
+    def __next__(self):
         """Retrieves the next resource."""
         # pylint: disable=next-method-called
         while not self._current:
-            self._current = self._iterator.next()
+            self._current = next(self._iterator)
```

Comment:
Python 3 style is for iterators to have a special method `__next__` and then invoke the built-in function `next()` to call it.

The `next()` method is retained as an alias only for Python 2 compatibility -- in Python 3, the iterator's next method is not meant to be invoked directly.

Renamed per your suggestion.